Os board module code

wuqiank opened the pull request:
@Richard
add os_board module code without external library into ossap_stats

wuqiank committed:
Remove the os_board part in this branch

wuqiank committed:
add os board

wuqiank committed:
add google client

wuqiank committed:
module code

rbran100 commented on ossap_stats/ossap-stats-embed-js.tpl.php:
```diff
@@ -63,7 +75,7 @@
     <?php $value = ($key != "os_version" && is_numeric($value)) ? number_format($value) : $value ?>
     <?php $value = ($key == 'activity-messages') ? $value : "'{$value}'" ?>
-    elements[i].innerHTML = <?php echo $value; ?>;
+	elements[i].innerHTML = <?php echo $value; ?>;
```
Looks like you replaced spaces with tabs here, can you set tabs to 2 spaces instead?

rbran100 commented on ossap_stats/ossap_stats.module:
```diff
@@ -31,10 +31,150 @@ function ossap_stats_menu() {
+function _os_board_get_require_once(){
+  require_once 'google-api-php-client/src/Google_Client.php';
+  require_once 'google-api-php-client/src/contrib/Google_AnalyticsService.php';
+  require_once 'os_board.inc';
```
Will this file exist?

rbran100 replied:
You can use: module_load_include()

rbran100 commented on ossap_stats/ossap_stats.module:
```diff
@@ -31,10 +31,150 @@ function ossap_stats_menu() {
   );
-
+  //osboard menu start
+  $items['ossap/most-viewed-pages'] = array(
```
Not sure you still need this menu entry?

rbran100 commented on ossap_stats/ossap_stats.module:
```diff
@@ -31,10 +31,150 @@ function ossap_stats_menu() {
+	'access callback' => TRUE,
+  );
+  
+  $items['ossap/most-visited-sites'] = array(
```
or this one?

rbran100 commented on ossap_stats/ossap_stats.module:
```diff
@@ -46,6 +186,9 @@ function ossap_stats_theme() {
         'messages' => NULL,
         'getsatisfaction' => NULL,
+		'os_board_visited_pages' => NULL,
+		'os_board_viewed_pages' => NULL
```
tabs again

rbran100 commented on ossap_stats/ossap_stats.module:
```diff
@@ -120,6 +276,9 @@ function _ossap_stats_embed_js($stat) {
     'getsatisfaction' => $getsatisfaction,
+	//added variables for most_visited_sites and most_viewed_pages
+	'most_visited_sites' => $most_visited_sites,
+	'most_viewed_pages' => $most_viewed_pages
```
tabs again, I will stop mentioning :-)

rbran100 commented on ossap_stats/ossap_stats_credentials.inc:
```diff
@@ -0,0 +1,127 @@
+* creates the google_client object
+* returns the validated service object
+*/
+function check_credentials(){
```
You need to namespace your function in here ex:

```
ossap_stats_check_credentials();
```

wuqiank committed:
tab remove

wuqiank committed:
remove tabs

wuqiank committed:
remove tabs

wuqiank committed:
use name space in function

priya-sudarshanam commented:
@Richard
add os_board module code without external library into ossap_stats

A little confused by the above:
Can you pls further elaborate on what it means

Thanks,
Priya

rbran100 commented:
@priya-sudarshanam I was saying that you can use https://api.drupal.org/api/drupal/includes!module.inc/function/module_load_include/7 module_load_include instead of require once

Also I didn't see os_board.inc in the Filesystem, are we sure that file exists?

wuqiank commented:
@rbran100 
We have replaced "require once" with "module_load_include" 
and we changed the .inc file name, "os_board.inc" was a wrong name of our existing .inc, so only one .inc file

Thanks a lot

priya-sudarshanam commented:
Hi Richard,
   Your instructions for the module_load_include was clear but I was confused on the following part:
"do you mind separating the external library and the module code into two separate PR's".

Yes you are right regarding the os_board.inc, it does not exist. It has now been renamed to ossap_stats_credentials.inc as Qi mentioned. The module_load_include
 is in and the function check_credentials has now been namespaced.

Thanks,
Priya

From: Richard Brandon [notifications@github.com]Sent: Thursday, May 01, 2014 5:00 PMTo: openscholar/ossapCc: Sudarshanam, PriyashreeSubject: Re: [ossap] Os board module code (#23)

@priya-sudarshanam I was saying that you can use

https://api.drupal.org/api/drupal/includes!module.inc/function/module_load_include/7 module_load_include instead of require once
Also I didn't see os_board.inc in the Filesystem, are we sure that file exists?
—
Reply to this email directly or 
view it on GitHub.
